src/com/android/settings/sudconfig/SudConfigProvider.kt

```diff
@@ -0,0 +1,97 @@
+package com.android.settings.sudconfig
+
+import android.os.Build
+import android.os.Bundle
+import android.provider.Settings
+import android.text.TextUtils
+import android.util.Log
+import com.android.settings.R;
+
+/**
+ * Provides system-wide config for setup wizard screens.
+ */
+class SudConfigProvider : NonRelationalProvider() {
+    companion object {
+        private const val TAG = "SudConfigProvider"
+
+        // keys inside overlay config bundle
+        private const val KEY_PACKAGE_NAME = "packageName"
+        private const val KEY_RESOURCE_NAME = "resourceName"
+        private const val KEY_RESOURCE_ID = "resourceId"
+
+        // methods
+        private const val METHOD_GET_SUW_DEFAULT_THEME_STRING = "suwDefaultThemeString"
+        const val METHOD_APPLY_GLIF_THEME_CONTROLLED_TRANSITION =
+            "applyGlifThemeControlledTransition"
+        const val METHOD_IS_DYNAMIC_COLOR_ENABLED = "isDynamicColorEnabled"
+        const val METHOD_IS_EMBEDDED_ACTIVITY_ONE_PANE_ENABLED = "isEmbeddedActivityOnePaneEnabled"
+        const val METHOD_IS_FULL_DYNAMIC_COLOR_ENABLED = "isFullDynamicColorEnabled"
+        const val METHOD_IS_MATERIAL_YOU_STYLE_ENABLED = "IsMaterialYouStyleEnabled"
+        const val METHOD_IS_NEUTRAL_BUTTON_STYLE_ENABLED = "isNeutralButtonStyleEnabled"
+        const val METHOD_IS_SUW_DAY_NIGHT_ENABLED = "isSuwDayNightEnabled"
+        const val METHOD_GET_DEVICE_NAME = "getDeviceName"
+        const val METHOD_GET_OVERLAY_CONFIG = "getOverlayConfig"
```

Comment:
Why declare single-use strings as separate constants? This makes the code harder to read, need to check two places to see how a method is handled. That's the reason why such strings weren't declared as separate constants in SetupDesignConfigProvider

Reply:
yes single use strings shouldn't be declared as constants, I don't prefer either.
but I had received comments previously by other members (on some other commits), so I thought maybe that's a standard being followed.
will remove these now, will be much better!